packages/ckeditor5-link/src/linkui.ts

```diff
@@ -195,28 +195,25 @@ export default class LinkUI extends Plugin {
 		const editor = this.editor;
 		const linkCommand: LinkCommand = editor.commands.get( 'link' )!;
 		const defaultProtocol = editor.config.get( 'link.defaultProtocol' );
-		const allowCreatingEmptyLinks = editor.config.get( 'link.allowCreatingEmptyLinks' );
 
-		const formView = new ( CssTransitionDisablerMixin( LinkFormView ) )( editor.locale, linkCommand );
+		const formView = new ( CssTransitionDisablerMixin( LinkFormView ) )( editor.locale, linkCommand, getFormValidators( editor ) );
 
 		formView.urlInputView.fieldView.bind( 'value' ).to( linkCommand, 'value' );
 
 		// Form elements should be read-only when corresponding commands are disabled.
 		formView.urlInputView.bind( 'isEnabled' ).to( linkCommand, 'isEnabled' );
 
-		// Disable the "save" button if the command is disabled or the input is empty despite being required.
-		formView.saveButtonView.bind( 'isEnabled' ).to(
-			linkCommand, 'isEnabled',
-			formView.urlInputView, 'isEmpty',
-			( isCommandEnabled, isInputEmpty ) => isCommandEnabled && ( allowCreatingEmptyLinks || !isInputEmpty )
-		);
+		// Disable the "save" button if the command is disabled.
+		formView.saveButtonView.bind( 'isEnabled' ).to( linkCommand, 'isEnabled' );
```

Comment:
> Correct me if I'm wrong, but I can't find a test that makes sure the submit button is always enabled and this can change only by disabling the command.

@oleq Exactly. I unified behavior in keyboard and mouse usage of form. It'd be kinda weird to allow to submit form in keyboard mode and on the other hand disable in mouse mode.

Reply:
@oleq It's tested right here: https://github.com/ckeditor/ckeditor5/blob/d784a5049612cd8c5d7265ca5afe7860c0c7b88c/packages/ckeditor5-link/tests/linkui.js#L145-L146